### PR TITLE
ci: release with tag

### LINKED ISSUE #16: build.pluginManagement.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.codehaus.mojo:versions-maven-plugin @ com._4paradigm.hybridse:hybridse-java:0.1.0-SNAPSHOT,
```log
Warning:  
Warning:  Some problems were encountered while building the effective model for com._4paradigm.hybridse:hybridse-common:jar:0.1.0-SNAPSHOT
Warning:  'build.pluginManagement.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.codehaus.mojo:versions-maven-plugin @ com._4paradigm.hybridse:hybridse-java:0.1.0-SNAPSHOT, /__w/HybridSE/HybridSE/java/pom.xml, line 50, column 25
Warning:  
Warning:  Some problems were encountered while building the effective model for com._4paradigm.hybridse:hybridse-java:pom:0.1.0-SNAPSHOT
Warning:  'build.pluginManagement.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.codehaus.mojo:versions-maven-plugin @ line 50, column 25
Warning:  
Warning:  It is highly recommended to fix these problems because they threaten the stability of your build.
Warning:  
Warning:  For this reason, future Maven versions might no longer support building such malformed projects.
```

### LINKED ISSUE #41: cicd: build on push events to default branch. release on tag push, upload artifacts on other commit push
**Is your feature request related to a problem? Please describe.**
extra steps on commit pushed to default branch
1. tag push: trigger a release build, with release note and asserts. 
   - tag may not at main, SemVer tag only
2. commit push to main: trigger a normal build, but upload build outputs as GitHub artifacts

**Describe the solution you'd like**
cicd
**Describe alternatives you've considered**
Nop
**Additional context**
Nop

TODO

- [x] auto release on tag push
- [ ] get release note from `CHANGELOG.md`
- [ ] 匹配代码中的 version
- [ ] upload asserts
  - [x]  linux archive
  - [ ] mac archive, require #47 
  - [ ] digest
  - [ ] digest.asc
- [ ] upload artifacts on commit push
  - note: artifacts take up space quota for private repo

### PR DESCRIPTION
**Pull request by aceforeverd**
+ resolve #41 , add a `release` step in GitHub action, which will create release on tag pushing.
+ resolve #16 
+ enable coverage build

### CODE REVIEW

**Comment by jingchen2222** [comment]
image
What's the problem of coverage test?

**Comment by aceforeverd** [comment]
> https://github.com/4paradigm/HybridSE/runs/2349939405?check_suite_focus=true
> 
> What's the problem of coverage test?

Marked. We should try on public host since the repo is open

**Review by jingchen2222 — APPROVED** [review]
LGTM

File: .github/workflows/hybridse-ci.yml [hunk 1]
```diff
@@ -58,6 +58,24 @@ jobs:
         check_name: Java Test Report
         comment_title: Java Test Report
 
+    - name: Create Archive
+      run: |
+        VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+        [[ "${{ github.ref }}" == "refs/tags/v"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+        echo "HYBRIDSE_VERSION=$VERSION" >> $GITHUB_ENV
+        HYBRIDSE_VERSION=$VERSION ./tools/hybridse_deploy.sh
```
[the remark this answers is not in the document]
Comment (aceforeverd) [reply .github/workflows/hybridse-ci.yml]: tracking at https://github.com/4paradigm/HybridSE/issues/41

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: tools/hybridse_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+if uname -a | grep -q Darwin; then
+    # in case coreutils not install on mac
+    alias nproc='sysctl -n hw.logicalcpu'
+fi
+
+mkdir -p build
+pushd build/
+
+cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX="hybridse"
```
[the remark this answers is not in the document]
Comment (aceforeverd) [reply tools/hybridse_deploy.sh:47]: 统一这里改了  #39

File: .github/workflows/hybridse-ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (aceforeverd) [reply .github/workflows/hybridse-ci.yml]: yep, but not tested

**Pull request merged**